src/App.jsx

```diff
@@ -14,45 +17,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function prepareGoods(goods, sortRule) {
+  const preperedGoods = [...goods];
+
+  if (sortRule && sortRule.includes('reverse')) {
+    const reversedArr = [...goods].sort(() => -1);
+
+    reversedArr.sort((good1, good2) => {
+      switch (sortRule) {
+        case 'reverse':
+          return 1;
+        case 'alphabetic reverse':
+          return good2.localeCompare(good1);
+        case 'length reverse':
+          return good1.includes(' ')
+            ? -1
+            : good2.length - good1.length;
+        default:
+          return 0;
+      }
+    });
+
+    return reversedArr;
+  }
+
+  if (sortRule) {
+    preperedGoods.sort((good1, good2) => {
+      switch (sortRule) {
+        case 'alphabetic':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.includes(' ')
+            ? 1
+            : good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preperedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const visibleGoods = prepareGoods(goodsFromServer, sortField);
+
+  const SORT_BY_NAME = 'alphabetic';
+  const SORT_BY_LENGTH = 'length';
+  const SORT_REVERSE = 'reverse';
+
+  const toggleReverse = () => setSortField((prevState) => {
+    if (prevState === '') {
+      return SORT_REVERSE;
+    }
+
+    const haveReverse = prevState.includes('reverse');
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    if (haveReverse && prevState.split(' ').length === 1) {
+      return '';
+    }
+
+    return haveReverse
+      ? prevState.split(' ')[0]
+      : `${sortField} ${SORT_REVERSE}`;
+  });
+
+  const handleSort = sortParameter => setSortField(prevSort => (
+    prevSort.includes(SORT_REVERSE)
+      ? `${sortParameter} ${SORT_REVERSE}`
+      : sortParameter));
+
+  const resetButton = sortField && (
+    <button
+      type="button"
+      className="button is-danger is-light"
+      onClick={() => setSortField('')}
```

Comment:
```suggestion
      onClick={reset}
```
It's better to create separate functions to handle events. This way, your code will be easier to maintain, extend, and reuse.

Imagine you'll have a situation where you need to reuse the same functionality on another event. Then you'd be able to just use the same function you've created. Also, if you need to extend the functionality of event handling in the future, it'd be easier to do so if you have a separate function. Moreover, if this function's used more than once you'll need to add new functionality only in one function and then call it wherever you need to.